### PR TITLE
Significantly reduce the number of SpotBugs warnings for the CLI

### PR DESCRIPTION
**Pull request by tristantarrant**

### CODE REVIEW

**Comment by tristantarrant** [comment]
Do we want Jiras for all PRs which reduce static analysis warnings?

File: cli/src/main/java/org/infinispan/cli/commands/Backup.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
    public static final String COUNTERS = "counters";
    public static final String PROTO_SCHEMAS = "proto-schemas";
    public static final String SCRIPTS = "scripts";
-   public static final String[] ALL_RESOURCE_TYPES = new String[]{CACHES, CACHE_CONFIGS, COUNTERS, PROTO_SCHEMAS, SCRIPTS};
+   static final String[] ALL_RESOURCE_TYPES = new String[]{CACHES, CACHE_CONFIGS, COUNTERS, PROTO_SCHEMAS, SCRIPTS};
```
Comment (pruivo) [inline cli/src/main/java/org/infinispan/cli/commands/Backup.java:43]: why not private?

File: cli/src/main/java/org/infinispan/cli/connection/rest/RestConnection.java [hunk 2]
```diff
@@ -100,7 +92,7 @@
  * @since 10.0
  **/
 public class RestConnection implements Connection, Closeable {
-   public static String PROTOBUF_METADATA_CACHE_NAME = "___protobuf_metadata";
+   public static final String PROTOBUF_METADATA_CACHE_NAME = "___protobuf_metadata";
```
Comment (pruivo) [inline cli/src/main/java/org/infinispan/cli/connection/rest/RestConnection.java:95]: private?

File: core/src/main/java/org/infinispan/configuration/serializing/XMLExtendedStreamWriterImpl.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
  * @since 9.0
  */
 public final class XMLExtendedStreamWriterImpl implements XMLExtendedStreamWriter, XMLStreamConstants {
-   private static final String NO_NAMESPACE = new String();
+   private static final String NO_NAMESPACE = "";
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/configuration/serializing/XMLExtendedStreamWriterImpl.java:19]: couple line below, `unspecifiedNamespaces` can be final.

**Comment by tristantarrant** [comment]
@pruivo I've addressed your comments and also added a commit to configure PMD correctly

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant !

**Pull request merged**